api/preprints/views.py

```diff
@@ -73,29 +73,37 @@
 class PreprintOldVersionsImmutableMixin:
     '''Override method to reject modify requests for old preprint versions (except for withdrawal)'''
     def update(self, request, *args, **kwargs):
-        preprint = self.get_preprint(check_object_permissions=False)
-        if preprint.is_latest_version or preprint.machine_state == 'initial':
-            return super().update(request, *args, **kwargs)
-        message = 'User can not edit previous versions of a preprint'
-        sentry.log_message(message)
-        raise Conflict(detail=message)
+        if hasattr(super(), 'update'):
+            preprint = self.get_preprint(check_object_permissions=False)
+            if preprint.is_latest_version or preprint.machine_state == 'initial':
+                return super().update(request, *args, **kwargs)
+            message = 'User can not edit previous versions of a preprint'
+            sentry.log_message(message)
+            raise Conflict(detail=message)
+        else:
+            raise MethodNotAllowed('update')
 
     def create(self, request, *args, **kwargs):
-        preprint = self.get_preprint(check_object_permissions=False)
-        if preprint.is_latest_version or preprint.machine_state == 'initial':
-            return super().create(request, *args, **kwargs)
-        message = 'User can not edit previous versions of a preprint'
-        sentry.log_message(message)
-        raise Conflict(detail=message)
+        if hasattr(super(), 'create'):
+            preprint = self.get_preprint(check_object_permissions=False)
+            if preprint.is_latest_version or preprint.machine_state == 'initial':
+                return super().create(request, *args, **kwargs)
+            message = 'User can not edit previous versions of a preprint'
+            sentry.log_message(message)
+            raise Conflict(detail=message)
+        else:
+            raise MethodNotAllowed('create')
 
     def delete(self, request, *args, **kwargs):
-        preprint = self.get_preprint(check_object_permissions=False)
-        if preprint.is_latest_version or preprint.machine_state == 'initial':
-            return super().delete(request, *args, **kwargs)
-        message = 'User can not edit previous versions of a preprint'
-        sentry.log_message(message)
-        raise Conflict(detail=message)
-
+        if hasattr(super(), 'delete'):
+            preprint = self.get_preprint(check_object_permissions=False)
+            if preprint.is_latest_version or preprint.machine_state == 'initial':
+                return super().delete(request, *args, **kwargs)
+            message = 'User can not edit previous versions of a preprint'
+            sentry.log_message(message)
+            raise Conflict(detail=message)
+        else:
+            raise MethodNotAllowed('delete')
 
```

Comment:
I will remove this commit since the `MethodNotAllowed` check is handled by DjangoAPI views, this has already been fixed in https://github.com/CenterForOpenScience/osf.io/pull/10888/commits/67904fbb3fd8228c09c3eb5f5d2b5b77eb374f7b